xchange-bithumb/src/main/java/org/knowm/xchange/bithumb/service/BithumbTradeService.java

```diff
@@ -96,9 +93,13 @@ public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
       } catch (BithumbException e) {
         throw BithumbErrorAdapter.adapt(e);
       }
+    } else if (orderParams instanceof CancelOrderByIdParams && orderParams instanceof CancelOrderByCurrencyPair) {
```

Comment:
`CancelOrderByIdParams` is meant to be used then you can cancel an order by just passing its id
`CancelOrderByCurrencyPair` i meant to be used then you can cancel all orders for a given currency pair

I don't think this is needed, just use `CancelOrderByPairAndIdParams`

besides this code doesn't do proper error handling

Reply:
I disagree with the premise her that interfaces a exclusive.  If this were the case then a client application has no way of knowing which variant of CancelOrderParams an exchange requires.

interface CancelOrderByCurrencyPair extends CancelOrderParams
interface CancelOrderByIdParams extends CancelOrderParams
interface CancelOrderByPairAndIdParams extends CancelOrderByIdParams 

these combinations are not consistent

One fix would be to:
public interface CancelOrderByPairAndIdParams extends CancelOrderByIdParams, CancelOrderByCurrencyPair {
}

however if we look at class TradeHistoryParamsAll,we see that these interfaces are clearly meant to be additive, rather than exclusive.
    implements TradeHistoryParamsTimeSpan,
        TradeHistoryParamPaging,
        TradeHistoryParamsIdSpan,
        TradeHistoryParamOffset,
        TradeHistoryParamCurrencyPair,
        TradeHistoryParamMultiCurrencyPair,
        TradeHistoryParamLimit 
since TradeHistoryParamCurrencyPair and TradeHistoryParamMultiCurrencyPair are mutually exclusive

NB: CancelOrderByPairAndIdParams is used by only two exchanges (bithumb and bx) and we are in the process of fixing bithumb

> besides this code doesn't do proper error handling
PR does no more than extend the existing error handling, if that is incorrect then it should be under another PR

Reply:
SimulatedTradeService:104
```
      if (orderParams instanceof CancelOrderByIdParams
          && orderParams instanceof CancelOrderByOrderTypeParams)
```
KrakenTradeService:60
```
    if (orderParams instanceof CancelOrderByIdParams) {
      return cancelOrder(((CancelOrderByIdParams) orderParams).getOrderId());
    }
    if (orderParams instanceof CancelOrderByUserReferenceParams) {
      return cancelOrder(((CancelOrderByUserReferenceParams) orderParams).getUserReference());
    }
```

Reply:
>  interfaces are clearly meant to be additive, rather than exclusive.

ok, your right